users.py

```diff
@@ -0,0 +1,113 @@
+"""User Signup and Signin class Logic"""
+
+from constant import (
+    ROLE,
+    ROLE_HOST,
+    ROLE_WATCHER,
+    USER_JSON,
+    USER_EXIST,
+    USER_NOT_EXIST,
+    WELCOME,
+    AUTHENTICATE_ERROR,
+    DATA_JSON_PATH,
+)
+from exceptions import AuthenticationError, UnderAge
+from utils import (
+    custom_print_msg,
+    read_json,
+    update_main_theater_data,
+    user_input,
+    user_input_int_type,
+    password_hash,
+)
+
+
+class User:
+    """__main__ class to implement user signup & signin process"""
+
+    def __init__(self, username: str) -> None:
+        """
+        user class Constructor
+        @param username:str
+        @param password:int
+        """
+        self.username = username
+        self.is_user_exist()
+
+    def sign_up(self):
+        """
+        method to implement signup process
+        @return:
+        """
+        custom_print_msg("Please Enter User Information")
+        password = password_hash(input("Enter Your Password: "))
+        role = user_input_int_type(f"Please Select User Role {ROLE}: ")
+        if int(role) == 1:
+            role = ROLE_HOST
+        else:
+            role = ROLE_WATCHER
+        name = user_input("Enter Name: ")
+        age = user_input_int_type("Enter Age: ")
+        if age < 18:
+            raise UnderAge("You are Under Age to Book Tickets Required 18 plus")
+        email = user_input("Enter Email: ")
+        status = self.insert_signup_information(password, name, int(age), email, role)
```

Comment:
can't you send kwargs ?, in future if I add or remove the fields then this functions args will get updated